### ISSUE TITLE
Refactor method signature (TypeScript)

### ISSUE THREAD

**Issue by LaP0573** [Microsoft/vscode#50276]
Webstorm (and Jetbrains IDEs in general) have this really neat refactoring functionality, allowing you to refactor a method's signature:

<img width="527" alt="screen shot 2018-05-22 at 14 10 52" src="https://user-images.githubusercontent.com/9264791/40361558-07ff140a-5dca-11e8-9a56-6d43f35d121d.png">

As you can see, you can remove/add fields, change types, names and so on, and the refactor will be repercuted on any usage of the method. This can be accessed by right clicking on the method and selecting Refactor > Change Signature.

You can also add (or remove) arguments to methods by changing how you call them. For example here I'm calling my constructor with a new argument (constructor initially takes no argument for `IdComponent`), yet VSCode is only pointing out it's an error, not offering me to change the signature and add this argument (of type number) to the constructor's signature.

<img width="532" alt="screen shot 2018-05-22 at 14 13 15" src="https://user-images.githubusercontent.com/9264791/40361633-520de634-5dca-11e8-93b1-4f66dfbc6f3f.png">

So basically, unless I've missed something, this would be very, very useful (and needed, refactoring code is a huge part of coding in general) :

- Allowing to change a method's signature by right clicking and refactoring it (add/remove arguments, change arguments names and types)
- Allowing to quick-add or remove arguments to a method call by simply calling it with more (or less) arguments than needed, and having an action automatically adding (or removing) the extra arguments.

**Comment by mjbvz** [comment]
This issue was moved to Microsoft/TypeScript#24355
